### DISCUSSION TITLE
Welcome to ResponsiveFramework Discussions!

### DISCUSSION THREAD

**Discussion by rayliverified** [Codelessly/ResponsiveFramework#35] [General]
Discuss the Responsive Framework here ⬇️

Feel free to leave your thought on the features, API, and functionality. Or use this to start a discussion and share what's on your mind.

**Comment by StrDragos** [comment]
Hey @searchy2, first I would thank you for this great library.  
Second I would like to propose a change to the ResposinveValue class, getter method for the value should not be an optional field since `ReactiveValue` already forces a `defaultValue`. 
The `defaultValue` is already checked : `value = getValue(context, conditions) ?? defaultValue;`
I would say that removing the possibility of the value being null will improve the ergonomics of this class usage. 
If we agree, I could do the PR myself.

**Reply by rayliverified** [reply]
Yes, please make a PR to help!

**Reply by StrDragos** [reply]
Hey, I believe I need permission to create branches or open PRs. 

```
ERROR: Permission to Codelessly/ResponsiveFramework.git denied to StrDragos.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Reply by rayliverified** [reply]
First, fork this repo and then you can propose a PR.

**Reply by StrDragos** [reply]
Done: https://github.com/Codelessly/ResponsiveFramework/pull/123

**Comment by Utkarsh4517** [comment]
Hey I just realized that this  project is requiring a wiki. I would love to create that. Should I go on?

**Reply by rayliverified** [reply]
What are you thinking about doing for the wiki?

**Reply by Utkarsh4517** [reply]
Sorry for the late reply, I was thinking about creating a wiki which consists step-by-step procedure about using and implementing this dependency with all of its properties, and some errors one might get while using Responsive framework. I will appreciate some suggestions from your side

**Comment by sahkaa** [comment]
Hey! Thanks for this plugin!
may I ask why ResponsiveBreakpoint.autoScale(600); was removed?

**Comment by rayliverified** [comment]
The legacy V1 API included autoScale. However, that did not work for users
who desired different breakpoints on different screens.
In the new V2, the autoScale behavior is separate. You can add autoScale
with ResponsiveScaledBox

https://github.com/Codelessly/ResponsiveFramework/blob/master/lib/src/responsive_scaled_box.dart

Example:
https://github.com/Codelessly/FlutterMinimalWebsite/blob/b776c89e269b1df637a6561f9b751fa0b102ab34/lib/main_advanced.dart
